### ISSUE TITLE
Implement search functionality in TranscriptContainer

### ISSUE THREAD

**Issue by allishultes** [bbc/digital-paper-edit-firebase#49]
This ticket can't be completed until after #45 #46 and #47 are finished.

**Is your feature request related to a problem? Please describe.**
After refactoring TranscriptContainer into a functional component, we need to re-implement searching functionality for the transcript, labels and annotations.

**Describe the solution you'd like**
- [x] The TranscriptContainer filters for text matching user input in real-time (i.e., before you press enter)
- [x] The Search element allows you to filter by speaker (you may need to review how speakers are added to the transcription, if this was not already done as part of an earlier ticket).
- [x] The Search element allows you to filter by label
- [x] The Search element allows you to filter by paragraph only

**Additional context**

This is a labelled view of search functionality: 
![search](https://user-images.githubusercontent.com/22618458/78792767-8ce75600-797f-11ea-92fb-6c6be00cd09b.png)

**Comment by emettely** [comment]
I'm going to start working on this and refactoring, so it's easier to pick up :)

**Comment by emettely** [comment]
See new search bar look:
<img width="522" alt="Screenshot 2020-04-28 at 14 18 49" src="https://user-images.githubusercontent.com/8188418/80450080-b2062d80-895b-11ea-9f0e-aab03ac09698.png">
<img width="526" alt="Screenshot 2020-04-28 at 14 18 37" src="https://user-images.githubusercontent.com/8188418/80450084-b3375a80-895b-11ea-928a-8d418c31d237.png">

**Comment by emettely** [comment]
Not all search functionality is stable. Some related to #80
